exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java

```diff
@@ -4,166 +4,178 @@
-	    System.out.println("They are player number " + players.size());
+
+	public boolean addPlayer(String playerName) {
+		players.add(new Player(playerName));
+		positionsOnBoard[numberOfPlayers()] = 0;
+		score[numberOfPlayers()] = 0;
+		inPenaltyBox[numberOfPlayers()] = false;
+
+		System.out.println(playerName + " was added");
+		System.out.println("They are player number " + players.size());
 		return true;
 	}
-	
-	public int howManyPlayers() {
+
+	public int numberOfPlayers() {
 		return players.size();
 	}
 
-	public void roll(int roll) {
-		System.out.println(players.get(currentPlayer) + " is the current player");
-		System.out.println("They have rolled a " + roll);
-		
+	public void playerTurn(int diceRoll) {
+		System.out.println(playerName() + " is the current player");
+		System.out.println("They have rolled a " + diceRoll);
+
 		if (inPenaltyBox[currentPlayer]) {
-			if (roll % 2 != 0) {
+			if (isOdd(diceRoll)) {
 				isGettingOutOfPenaltyBox = true;
-				
-				System.out.println(players.get(currentPlayer) + " is getting out of the penalty box");
-				places[currentPlayer] = places[currentPlayer] + roll;
-				if (places[currentPlayer] > 11) places[currentPlayer] = places[currentPlayer] - 12;
-				
-				System.out.println(players.get(currentPlayer) 
-						+ "'s new location is " 
-						+ places[currentPlayer]);
-				System.out.println("The category is " + currentCategory());
+				System.out.println(playerName() + " is getting out of the penalty box");
+
+				movePlayer(diceRoll);
+				displayPlayerPosition();
 				askQuestion();
 			} else {
-				System.out.println(players.get(currentPlayer) + " is not getting out of the penalty box");
+				System.out.println(playerName() + " is not getting out of the penalty box");
 				isGettingOutOfPenaltyBox = false;
-				}
-			
+			}
+
 		} else {
-		
-			places[currentPlayer] = places[currentPlayer] + roll;
-			if (places[currentPlayer] > 11) places[currentPlayer] = places[currentPlayer] - 12;
-			
-			System.out.println(players.get(currentPlayer) 
-					+ "'s new location is " 
-					+ places[currentPlayer]);
-			System.out.println("The category is " + currentCategory());
+			movePlayer(diceRoll);
+			displayPlayerPosition();
 			askQuestion();
 		}
-		
+	}
+
+	private boolean isOdd(int diceRoll) {
+		return diceRoll % 2 != 0;
+	}
+
+	private void displayPlayerPosition() {
+		System.out.println(playerName()
+				+ "'s new location is "
+				+ positionsOnBoard[currentPlayer]);
+	}
+
+	private void movePlayer(int diceRoll) {
+		positionsOnBoard[currentPlayer] = positionsOnBoard[currentPlayer] + diceRoll;
+		if (positionsOnBoard[currentPlayer] > 11) {
+			returnToBeginningOfBoard();
+		}
+		//TODO: finish implementing move player
+//        players.get(currentPlayer).getCurrentGameBoardPosition() + diceRoll;
+	}
+
+	private void returnToBeginningOfBoard() {
+		positionsOnBoard[currentPlayer] = positionsOnBoard[currentPlayer] - TOTAL_NUMBER_OF_BOARD_SQUARES;
 	}

```

Comment:
nice one 👍 like how you extracted methods and made small ones with single functionality. Helps to read it better